book/src/pil/expressions.md

```diff
@@ -46,28 +46,34 @@ Parentheses are allowed at any point to force precedence.
 A ``{``-``}``-delimited block can be used everywhere where an expression is expected.
 
 It has the form ``{ <statement> ; <statement> ; ... ; <expression> }``,
-i.e. a sequence of statements followed by an expression and the value of the statement block
-is the value of the final expression.
+i.e. a sequence of statements followed by an expression.
+The statements can either be expressions or let statements: ``let x = ...`` / ``let x;``
```

Comment:
what do non-let expressions do in that case if they're not the last statement?

Reply:
A bit further down:
```
Similarly, an expression at statement level (e.g. ``x * (x - 1) = 0;``) can be used to create new constraints that are added to the global constraint set
and this can only be done inside a [``constr``-functions](#constr-and-query-functions).
```

Reply:
ah right